src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90

```diff
@@ -1548,7 +1548,461 @@ subroutine read_obs_sm_ASCAT(                                  &
+       
+       do kk = 1,N_files
+          
+          ! open bufr file
+          
+          call closbf(lnbufr)  ! if a file with unit number lnbufr is open in (or "linked" with) BUFR, close it
+          open(lnbufr, file=trim(fnames(kk)), action='read', form='unformatted')
+          call openbf(lnbufr, 'SEC3', lnbufr) 
+          call mtinfo( trim(this_obs_param%path) // '/BUFR_mastertable/', lnbufr+1, lnbufr+2)
+          call datelen(10)
+          
+          msg_report: do while( ireadmg(lnbufr,subset,idate) == 0 )
+             
+             loop_report: do while( ireadsb(lnbufr) == 0 )
+                
+                ! columns of tmp_data:                  1    2    3    4    5    6    7    8    9    10   11   12   13   14    15
+
+                call ufbint(lnbufr,tmp_vdata,15,1,iret,'YEAR MNTH DAYS HOUR MINU SECO SSOM DOMO SMPF SMCF ALFR TPCX IWFR CLATH CLONH')
+
+                N_obs = N_obs + 1
+
+                tmp_data(N_obs,:) = tmp_vdata
+                
+             end do loop_report
+          end do msg_report
+          
+          call closbf(lnbufr)
+          close(lnbufr)
+          
+       end do ! end file loop
+
+       ! ----------------------------------------------------------------
+       !
+       ! select obs within assimilation window and from desired orbit direction; apply basic QC based on obs info
+       
+       N_tmp = 0 
+       
+       do kk = 1,N_obs
+          
+          date_time_tmp%year  = int(tmp_data(kk, 1))
+          date_time_tmp%month = int(tmp_data(kk, 2))
+          date_time_tmp%day   = int(tmp_data(kk, 3))
+          date_time_tmp%hour  = int(tmp_data(kk, 4))
+          date_time_tmp%min   = int(tmp_data(kk, 5))
+          date_time_tmp%sec   = int(tmp_data(kk, 6))
+          
+          ! skip if record outside of current assim window
+          if ( datetime_lt_refdatetime( date_time_tmp, date_time_low ) .and.           &
+               datetime_le_refdatetime( date_time_up, date_time_tmp )         ) cycle
+          
+          ! skip if record contains invalid soil moisture value
+          if ( tmp_data(kk, 7) > 100. .or. tmp_data(kk, 7) < 0. ) cycle     
+          
+          ! EUMETSAT file contains data of both ascending and descending half-orbits. 
+          ! DOMO ("Direction of motion of moving observing platform") is used to separate Asc and Desc
+          !   because the file doesn't contain any explicit orbit indicator variable.
+          ! According to Pamela Schoebel-Pattiselanno, EUMETSAT User Services Helpdesk:
+          !   "When the value (of DOMO) is between 180 and 270 degrees, it is the descending part 
+          !    of the orbit.  When it is between 270 and 360 degrees, it is the ascending part."
+          if (index(this_obs_param%descr,'_A') /=0 .and. (tmp_data(kk, 8) < 270 .or. tmp_data(kk, 8) >  360)) cycle
+          if (index(this_obs_param%descr,'_D') /=0 .and. (tmp_data(kk, 8) < 180 .or. tmp_data(kk, 8) >= 270)) cycle
```

Comment:
Maybe let's revisit the separation of the ASCAT sm species into Asc and Desc.  In the past, we haven't really used this distinction in the LDAS for anything, and it just doubles the total cost of the BUFR read.